src/components/ClassStudentList/ClassStudentList.tsx

```diff
@@ -95,6 +98,7 @@ const ClassStudentList = ({ classObject }: ClassStudentListProps) => {
                 key={createUniqueStudentKey(student, index)}
                 student={student}
                 openEditModal={handleOpenEditStudentModal}
+                openUploadLetterModal={() => openUploadLetterModal(student)}
```

Comment:
Shouldn't call functions in the declaration of a component. Pass the down the function and call it with the parameter within the compenent.

```tsx
// change this line to:
 openUploadLetterModal={openUploadLetterModal}
 
 // in the component
() => openUploadLetterModal(student)
```